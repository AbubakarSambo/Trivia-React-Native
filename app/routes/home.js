import React, { Component } from 'react'
import { connect } from 'react-redux'
import { StyleSheet, Text, View, TouchableOpacity } from 'react-native'
import {getQuestions} from '../actions/questions'

export class Home extends Component {
  componentDidMount () {
    this.props.getQuestions()
  }

  render () {
    return (
      <View style={styles.container}>
        <View style={styles.header}>
          <Text style={styles.headerText}>Welcome to the Trivia Challenge</Text>
        </View>
        <View style={styles.body}>
          <View >
            <Text style={styles.bodyText}>Your mission should you choose to accept it </Text>
            <Text style={styles.bodyText}>will be to answer 10 True/False questions </Text>
          </View>
          <Text style={styles.score}>Can you score 100%</Text>
        </View>
        <View style={styles.footer}>
          <TouchableOpacity style={styles.begin} onPress={() => this.props.navigation.navigate('Quiz')}>
            <Text style={{ color: '#fff' }}>Begin</Text>
          </TouchableOpacity>
        </View>
      </View>
    )
  }
}

function mapDispatchToProps (dispatch) {
  return {
    getQuestions: () => dispatch(getQuestions())
  }
}

export default connect(null, mapDispatchToProps)(Home)

const styles = StyleSheet.create({
  container: {
    display: 'flex',
    margin: '1%',
    flex: 1,
    paddingTop: '15%'
  },
  headerText: {
    fontSize: 20
  },
  bodyText: {
    fontSize: 15
  },
  score: {
    fontSize: 25
  },
  header: {
    flexGrow: 1,
    alignItems: 'center',
    justifyContent: 'center'
  },
  body: {
    flexGrow: 1,
    alignItems: 'center',
    justifyContent: 'space-between'
  },
  footer: {
    flexGrow: 1,
    justifyContent: 'center',
    alignItems: 'center'
  },
  begin: {
    borderRadius: 10,
    backgroundColor: '#68a0cf',
    paddingTop: 10,
    paddingBottom: 10,
    width: 100,
    alignItems: 'center'

  }
})
