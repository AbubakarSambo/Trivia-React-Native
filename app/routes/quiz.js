import React, { Component } from 'react';
import { connect } from 'react-redux';
import { Icon } from 'react-native-elements'
import entities from 'entities'
import { StyleSheet, Text, View, ActivityIndicator, WebView } from 'react-native';
import { setQuestions } from '../actions/questions'


export class Quiz extends Component {
  constructor(props) {
    super(props)
    this.state = {
      questionIndex: 0
    }
  }
  answered = (bool) => {
    let { questionIndex } = this.state
    let { questions } = this.props

    let updatedQuestions = questions.map((item, index) => {
      if (index === questionIndex) {
        return { ...item, selectedAnswer: bool }
      }
      return item
    })

    this.props.setQuestions(updatedQuestions)
    if (questionIndex + 1 === questions.length) {
      let totalScore = updatedQuestions.reduce((total, question) => {
        let value = 0
        if (question.correct_answer === question.selectedAnswer) {
          value = 1
        }
        return total + value
      }, 0)
      this.props.navigation.navigate('Results', {
        totalScore
      })
    }
    else {
      this.setState({ questionIndex: questionIndex + 1 })
    }
  }
  render() {
    const { questions } = this.props
    let { questionIndex } = this.state

    return (

      <View style={styles.container}>
        {this.props.loading ?
          <ActivityIndicator style={styles.loader} size="large" color="#000" /> :
          <View>
            <View style={styles.header}>
              <Text style={styles.headerText}>{questions.length > 0 && questions[questionIndex].category}</Text>
            </View>
            <View style={styles.body}>
              <View style={styles.questionBody}>
                <Text>{entities.decodeHTML(questions[questionIndex].question)}</Text>
              </View>
              <View >
                <Text style={styles.questionNumber}>{questionIndex + 1} out of {questions.length}</Text>
              </View>
            </View>
            <View style={styles.footer}>
              <View style={styles.selectAnswer}>
                <Icon
                  raised
                  name='times'
                  type='font-awesome'
                  color='#f50'
                  onPress={() => this.answered('False')} />
                <Text style={styles.bodyText}>False</Text>
              </View>
              <View style={styles.selectAnswer}>
                <Icon
                  raised
                  name='check'
                  type='font-awesome'
                  color='#00FF00'
                  onPress={() => this.answered('True')} />
                <Text style={styles.bodyText}>True</Text>
              </View>
            </View>
          </View>
        }
      </View>
    );
  }
}

function mapStateToProps(state) {
  return {
    questions: state.questions,
    loading: state.loading
  }
}
function mapDispatchToProps(dispatch) {
  return {
    setQuestions: (questions) => dispatch(setQuestions(questions))
  }
}

export default connect(mapStateToProps, mapDispatchToProps)(Quiz)

const styles = StyleSheet.create({
  container: {
    display: 'flex',
    margin: '1%',
    flex: 1,
    paddingTop: '15%',
  },
  headerText: {
    fontSize: 20
  },
  header: {
    flexGrow: 1,
    alignItems: 'center',
    justifyContent: 'center'
  },
  body: {
    flexGrow: 1,
  },
  questionBody: {
    margin: 50,
    height: 100,
    borderWidth: 2,
    padding: 10
  },
  loader: {
    display: 'flex',
    justifyContent: 'center',
    alignItems: 'center'
  },
  questionNumber: {
    textAlign: 'center',
    fontSize: 20
  },
  bodyText: {
    fontSize: 20
  },
  footer: {
    flexGrow: 1,
    justifyContent: 'space-between',
    alignItems: 'center',
    flexDirection: 'row',
    margin: 50
  },
  selectAnswer: {
    display: 'flex',
    alignItems: 'center'
  }
});
