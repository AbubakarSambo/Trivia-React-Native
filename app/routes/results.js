import React, { Component } from 'react';
import { connect } from 'react-redux'
import { StyleSheet, Text, View, TouchableOpacity, FlatList } from 'react-native'
import { getQuestions } from '../actions/questions'
import { ResultRow } from '../components/resultrow'

class Results extends Component {

  goHome = () =>{
   this.props.getQuestions()
   this.props.navigation.navigate('Home')
  }
  
  render() {
    const { navigation, questions } = this.props;
    return (
      <View style={styles.container}>
        <View style={styles.header}>
          <Text style={styles.headerText}>You Scored</Text>
          <Text style={styles.headerText}>{navigation.getParam('totalScore', 'Not Available')}/{questions.length}</Text>
        </View>
        <View style={styles.body}>
          
          <FlatList
            data={questions}
            renderItem={({ item }) => (
              <ResultRow item={item} />
            )}
            keyExtractor={(item, index) => index.toString()}
          />
        </View>
        <View style={styles.footer}>
          <TouchableOpacity style={styles.restart} onPress={() => this.goHome()}>
            <Text style={{ color: '#fff' }}>Play Again</Text>
          </TouchableOpacity>
        </View>
      </View>
    );
  }
}
function mapStateToProps(state) {
  return {
    questions: state.questions
  }
}
function mapDispatchToProps(dispatch) {
  return {
    getQuestions: () => dispatch(getQuestions())
  }
}

export default connect(mapStateToProps, mapDispatchToProps)(Results)

const styles = StyleSheet.create({
  container: {
    display: 'flex',
    marginTop: '10%',
    flex: 1,
  },
  headerText: {
    fontSize: 20
  },
  header: {
    flexGrow: 1,
    alignItems: 'center',
    justifyContent: 'center',
  },
  body: {
    flexGrow: 1,
    flexDirection: 'column',
    display: 'flex',
    height: 250

  },
  footer: {
    flexGrow: 1,
    justifyContent: 'center',
    alignItems: 'center',

  },
  restart: {
    borderRadius: 10,
    backgroundColor: '#68a0cf',
    paddingTop: 10,
    paddingBottom: 10,
    width: 100,
    alignItems: 'center',
  }
});
