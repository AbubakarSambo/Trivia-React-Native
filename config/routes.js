import { createStackNavigator } from 'react-navigation'

import Home from '../app/routes/home'
import Quiz from '../app/routes/quiz'
import Results from '../app/routes/results'

export default Routes = createStackNavigator({
  Home: { screen: Home },
  Quiz: { screen: Quiz },
  Results: { screen: Results }
},
{
  headerMode: 'none',
  navigationOptions: {
    headerVisible: false
  }
}
)
