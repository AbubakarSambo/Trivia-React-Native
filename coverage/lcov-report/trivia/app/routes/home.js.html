<!doctype html>
<html lang="en">
<head>
    <title>Code coverage report for trivia/app/routes/home.js</title>
    <meta charset="utf-8" />
    <link rel="stylesheet" href="../../../prettify.css" />
    <link rel="stylesheet" href="../../../base.css" />
    <meta name="viewport" content="width=device-width, initial-scale=1">
    <style type='text/css'>
        .coverage-summary .sorter {
            background-image: url(../../../sort-arrow-sprite.png);
        }
    </style>
</head>
<body>
<div class='wrapper'>
  <div class='pad1'>
    <h1>
      <a href="../../../index.html">All files</a> / <a href="index.html">trivia/app/routes</a> home.js
    </h1>
    <div class='clearfix'>
      <div class='fl pad1y space-right2'>
        <span class="strong">83.33% </span>
        <span class="quiet">Statements</span>
        <span class='fraction'>5/6</span>
      </div>
      <div class='fl pad1y space-right2'>
        <span class="strong">100% </span>
        <span class="quiet">Branches</span>
        <span class='fraction'>0/0</span>
      </div>
      <div class='fl pad1y space-right2'>
        <span class="strong">80% </span>
        <span class="quiet">Functions</span>
        <span class='fraction'>4/5</span>
      </div>
      <div class='fl pad1y space-right2'>
        <span class="strong">83.33% </span>
        <span class="quiet">Lines</span>
        <span class='fraction'>5/6</span>
      </div>
    </div>
  </div>
  <div class='status-line high'></div>
<pre><table class="coverage">
<tr><td class="line-count quiet">1
2
3
4
5
6
7
8
9
10
11
12
13
14
15
16
17
18
19
20
21
22
23
24
25
26
27
28
29
30
31
32
33
34
35
36
37
38
39
40
41
42
43
44
45
46
47
48
49
50
51
52
53
54
55
56
57
58
59
60
61
62
63
64
65
66
67
68
69
70
71
72
73
74
75
76
77
78
79
80
81
82
83
84
85
86
87
88</td><td class="line-coverage quiet"><span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-yes">1x</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-yes">1x</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-no">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-yes">1x</span>
<span class="cline-any cline-yes">1x</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-yes">1x</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span></td><td class="text"><pre class="prettyprint lang-js">import React, { Component } from 'react';
import { connect } from 'react-redux';
import { StyleSheet, Text, View, TouchableOpacity } from 'react-native';
import {getQuestions} from '../../actions/questions'
&nbsp;
&nbsp;
export class Home extends Component {
  componentWillMount(){
    this.props.getQuestions()
  }
  
  render() {
    return (
      &lt;View style={styles.container}&gt;
        &lt;View style={styles.header}&gt;
          &lt;Text style={[styles.font, styles.headerText]}&gt;Welcome to the Trivia Challenge&lt;/Text&gt;
        &lt;/View&gt;
        &lt;View style={styles.body}&gt;
          &lt;View &gt;
            &lt;Text style={[styles.font, styles.bodyText]}&gt;Your mission should you choose to accept it &lt;/Text&gt;
            &lt;Text style={[styles.font, styles.bodyText]}&gt;will be to answer 10 True/False questions &lt;/Text&gt;
          &lt;/View&gt;
          &lt;Text style={[styles.font, styles.score]}&gt;Can you score 100%&lt;/Text&gt;
        &lt;/View&gt;
        &lt;View style={styles.footer}&gt;
          &lt;TouchableOpacity style={styles.begin} onPress={<span class="fstat-no" title="function not covered" >()</span> =&gt; <span class="cstat-no" title="statement not covered" >this.props.navigation.navigate('Quiz')}</span>&gt;
              &lt;Text style={{ color: '#fff' }}&gt;Begin&lt;/Text&gt;
          &lt;/TouchableOpacity&gt;
        &lt;/View&gt;
      &lt;/View&gt;
    );
  }
}
&nbsp;
&nbsp;
function mapDispatchToProps(dispatch) {
  return {
    getQuestions: () =&gt; dispatch(getQuestions())
  }
}
&nbsp;
export default connect(null, mapDispatchToProps)(Home)
&nbsp;
const styles = StyleSheet.create({
  container: {
    display: 'flex',
    margin: '1%',
    flex: 1,
    paddingTop: '15%'
  },
  font: {
    fontFamily: 'Roboto',
  },
  headerText: {
    fontSize: 20
  },
  bodyText: {
    fontSize: 15
  },
  score: {
    fontSize: 25
  },
  header: {
    flexGrow: 1,
    alignItems: 'center',
    justifyContent: 'center'
  },
  body: {
    flexGrow: 1,
    alignItems: 'center',
    justifyContent: 'space-between'
  },
  footer: {
    flexGrow: 1,
    justifyContent: 'center',
    alignItems: 'center',
  },
  begin: {
    borderRadius: 10,
    backgroundColor: '#68a0cf',
    paddingTop: 10,
    paddingBottom: 10,
    width: 100,
    alignItems: 'center',
&nbsp;
  }
});
&nbsp;</pre></td></tr>
</table></pre>
<div class='push'></div><!-- for sticky footer -->
</div><!-- /wrapper -->
<div class='footer quiet pad2 space-top1 center small'>
  Code coverage
  generated by <a href="https://istanbul.js.org/" target="_blank">istanbul</a> at Mon Sep 17 2018 19:07:54 GMT+0100 (WAT)
</div>
</div>
<script src="../../../prettify.js"></script>
<script>
window.onload = function () {
        if (typeof prettyPrint === 'function') {
            prettyPrint();
        }
};
</script>
<script src="../../../sorter.js"></script>
</body>
</html>
