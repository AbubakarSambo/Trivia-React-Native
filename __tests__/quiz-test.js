import React from 'react'
import renderer from 'react-test-renderer'
import { Quiz } from '../app/routes/quiz'

it('renders Quiz page', () => {
  let questions = [
    {
      'category': 'Entertainment: Video Games',
      'type': 'boolean',
      'difficulty': 'hard',
      'question': 'Unturned originally started as a Roblox game.',
      'correct_answer': 'True',
      'incorrect_answers': [
        'False'
      ]
    },
    {
      'category': 'Science',
      'type': 'boolean',
      'difficulty': 'hard',
      'question': 'is pluto a planet',
      'correct_answer': 'False',
      'incorrect_answers': [
        'True'
      ]
    }
  ]
  const rendered = renderer.create(<Quiz questions={questions} />).toJSON()
  expect(rendered).toMatchSnapshot()
})
