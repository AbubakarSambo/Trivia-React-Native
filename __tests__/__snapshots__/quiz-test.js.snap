// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders Quiz page 1`] = `
<View
  style={
    Object {
      "display": "flex",
      "flex": 1,
      "margin": "1%",
      "paddingTop": "15%",
    }
  }
>
  <View>
    <View
      style={
        Object {
          "alignItems": "center",
          "flexGrow": 1,
          "justifyContent": "center",
        }
      }
    >
      <Text
        accessible={true}
        allowFontScaling={true}
        ellipsizeMode="tail"
        style={
          Array [
            Object {
              "fontFamily": "Roboto",
            },
            Object {
              "fontSize": 20,
            },
          ]
        }
      >
        Entertainment: Video Games
      </Text>
    </View>
    <View
      style={
        Object {
          "flexGrow": 1,
        }
      }
    >
      <View
        style={
          Object {
            "borderWidth": 2,
            "height": 100,
            "margin": 50,
          }
        }
      >
        <View
          style={
            Object {
              "flex": 1,
            }
          }
        >
          <RCTWebView
            allowsInlineMediaPlayback={undefined}
            automaticallyAdjustContentInsets={undefined}
            bounces={undefined}
            contentInset={undefined}
            dataDetectorTypes={undefined}
            decelerationRate={undefined}
            injectedJavaScript={undefined}
            mediaPlaybackRequiresUserAction={undefined}
            messagingEnabled={false}
            onLoadingError={[Function]}
            onLoadingFinish={[Function]}
            onLoadingStart={[Function]}
            onMessage={[Function]}
            onShouldStartLoadWithRequest={undefined}
            scalesPageToFit={true}
            scrollEnabled={undefined}
            source={
              Object {
                "html": "Unturned originally started as a Roblox game.",
              }
            }
            style={
              Array [
                Object {
                  "flex": 1,
                },
                Object {
                  "backgroundColor": "#ffffff",
                },
                undefined,
              ]
            }
          />
        </View>
      </View>
      <View>
        <Text
          accessible={true}
          allowFontScaling={true}
          ellipsizeMode="tail"
          style={
            Object {
              "fontSize": 20,
              "textAlign": "center",
            }
          }
        >
          1
           out of 
          2
        </Text>
      </View>
    </View>
    <View
      style={
        Object {
          "alignItems": "center",
          "flexDirection": "row",
          "flexGrow": 1,
          "justifyContent": "space-between",
          "margin": 50,
        }
      }
    >
      <View
        style={
          Object {
            "alignItems": "center",
            "display": "flex",
          }
        }
      >
        <View
          accessibilityComponentType={undefined}
          accessibilityLabel={undefined}
          accessibilityTraits={undefined}
          accessible={true}
          hasTVPreferredFocus={undefined}
          hitSlop={undefined}
          isTVSelectable={true}
          nativeID={undefined}
          onLayout={undefined}
          onResponderGrant={[Function]}
          onResponderMove={[Function]}
          onResponderRelease={[Function]}
          onResponderTerminate={[Function]}
          onResponderTerminationRequest={[Function]}
          onStartShouldSetResponder={[Function]}
          style={
            Array [
              Object {
                "margin": 7,
              },
              Object {
                "borderRadius": 28,
                "height": 52,
                "width": 52,
              },
              Object {
                "shadowColor": "rgba(0,0,0, .4)",
                "shadowOffset": Object {
                  "height": 1,
                  "width": 1,
                },
                "shadowOpacity": 1,
                "shadowRadius": 1,
              },
              Object {
                "alignItems": "center",
                "backgroundColor": "white",
                "justifyContent": "center",
              },
              undefined,
            ]
          }
          testID={undefined}
          tvParallaxProperties={undefined}
        >
          <Text
            accessible={true}
            allowFontScaling={false}
            ellipsizeMode="tail"
            style={
              Array [
                Object {
                  "color": "#f50",
                  "fontSize": 24,
                },
                Array [
                  Object {
                    "backgroundColor": "transparent",
                  },
                  undefined,
                ],
                Object {
                  "fontFamily": "FontAwesome",
                  "fontStyle": "normal",
                  "fontWeight": "normal",
                },
              ]
            }
          >
            
          </Text>
        </View>
        <Text
          accessible={true}
          allowFontScaling={true}
          ellipsizeMode="tail"
          style={
            Object {
              "fontFamily": "Roboto",
              "fontSize": 20,
            }
          }
        >
          False
        </Text>
      </View>
      <View
        style={
          Object {
            "alignItems": "center",
            "display": "flex",
          }
        }
      >
        <View
          accessibilityComponentType={undefined}
          accessibilityLabel={undefined}
          accessibilityTraits={undefined}
          accessible={true}
          hasTVPreferredFocus={undefined}
          hitSlop={undefined}
          isTVSelectable={true}
          nativeID={undefined}
          onLayout={undefined}
          onResponderGrant={[Function]}
          onResponderMove={[Function]}
          onResponderRelease={[Function]}
          onResponderTerminate={[Function]}
          onResponderTerminationRequest={[Function]}
          onStartShouldSetResponder={[Function]}
          style={
            Array [
              Object {
                "margin": 7,
              },
              Object {
                "borderRadius": 28,
                "height": 52,
                "width": 52,
              },
              Object {
                "shadowColor": "rgba(0,0,0, .4)",
                "shadowOffset": Object {
                  "height": 1,
                  "width": 1,
                },
                "shadowOpacity": 1,
                "shadowRadius": 1,
              },
              Object {
                "alignItems": "center",
                "backgroundColor": "white",
                "justifyContent": "center",
              },
              undefined,
            ]
          }
          testID={undefined}
          tvParallaxProperties={undefined}
        >
          <Text
            accessible={true}
            allowFontScaling={false}
            ellipsizeMode="tail"
            style={
              Array [
                Object {
                  "color": "#00FF00",
                  "fontSize": 24,
                },
                Array [
                  Object {
                    "backgroundColor": "transparent",
                  },
                  undefined,
                ],
                Object {
                  "fontFamily": "FontAwesome",
                  "fontStyle": "normal",
                  "fontWeight": "normal",
                },
              ]
            }
          >
            
          </Text>
        </View>
        <Text
          accessible={true}
          allowFontScaling={true}
          ellipsizeMode="tail"
          style={
            Object {
              "fontFamily": "Roboto",
              "fontSize": 20,
            }
          }
        >
          True
        </Text>
      </View>
    </View>
  </View>
</View>
`;
